pages/api/webhooks/customer-created.ts

```diff
@@ -1,18 +1,26 @@
-import { SALEOR_DOMAIN_HEADER } from "@saleor/app-sdk/const";
-import { withSaleorEventMatch, withWebhookSignatureVerified } from "@saleor/app-sdk/middleware";
+import { SALEOR_API_URL_HEADER } from "@saleor/app-sdk/const";
+import { SaleorAsyncWebhook } from "@saleor/app-sdk/handlers/next";
 import { withSentry } from "@sentry/nextjs";
 import type { Handler } from "retes";
 import { toNextHandler } from "retes/adapter";
 import { Response } from "retes/response";
 
 import Klaviyo from "../../../lib/klaviyo";
 import { getValue } from "../../../lib/metadata";
-import { withSaleorDomainMatch } from "../../../lib/middlewares";
+import { saleorApp } from "../../../saleor-app";
+
+export const customerCreatedWebhook = new SaleorAsyncWebhook<unknown>({
+  name: "Customer Created",
+  webhookPath: "api/webhooks/customer-created",
+  asyncEvent: "CUSTOMER_CREATED",
+  apl: saleorApp.apl,
+  query: "{}",
+});
 
 const handler: Handler = async (request) => {
-  const saleorDomain = request.headers[SALEOR_DOMAIN_HEADER];
-  const klaviyoToken = await getValue(saleorDomain as string, "PUBLIC_TOKEN");
-  const klaviyoMetric = await getValue(saleorDomain as string, "CUSTOMER_CREATED_METRIC");
+  const saleorApiUrl = request.headers[SALEOR_API_URL_HEADER];
```

Comment:
- retes can be dropped
- since you are using `createHandler` the API is available at `context.authData.saleorApiUrl`